jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPAProcessInstanceManager.java

```diff
@@ -174,6 +174,7 @@ public ProcessInstance getProcessInstance(long id, boolean readOnly) {
                 ((ProcessInstanceImpl) processInstance).reconnect();
                 if (readOnly) {
                     internalRemoveProcessInstance(processInstance);
+                    ((ProcessInstanceImpl) processInstance).setReadOnly();
```

Comment:
Doesn't the `setReadOnly` method execute exactly the same method as `internalRemoveProcessInstance(processInstance);`? In `setReadOnly` method there is `...getProcessInstanceManager().internalRemoveProcessInstance(this);` The only addition I see is setting the `process` field to null in the `setReadOnly` method.

Reply:
what is your point ?

Reply:
Sorry, maybe the comment was confusing.

I am saying that `setReadOnly()` method calls:

```
public void setReadOnly() {
        ((InternalProcessRuntime) kruntime.getProcessRuntime()).getProcessInstanceManager().internalRemoveProcessInstance(this);
        process = null;
}
```

which is (specifically `internalRemoveProcessInstance(this)`) already called one line above in the code I commented:

```
if (readOnly) {
           internalRemoveProcessInstance(processInstance); // this is duplicated imho
          ((ProcessInstanceImpl) processInstance).setReadOnly();
```

And since the implementation of `internalRemoveProcessInstance()` is:

https://github.com/kiegroup/jbpm/blob/de4094d0023fe11e6188fa86fea85f85c0cddfc3/jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPAProcessInstanceManager.java#L203-L205

Then I think the only real addition made by the new `setReadOnly` method is that it does `process = null;`. So why not only call the new `setReadOnly()` method in that if block?

But maybe there is something more to it which I didn't notice.

Reply:
my bad... I made a mistake with the latest push and the some changes were missing ..